modules/imgproc/src/hough.cpp

```diff
@@ -184,16 +199,36 @@ HoughLinesStandard( const Mat& img, float rho, float theta,
 
     // stage 4. store the first min(total,linesMax) lines to the output buffer
     linesMax = std::min(linesMax, (int)_sort_buf.size());
+
+    // create lines
     double scale = 1./(numrho+2);
-    for( i = 0; i < linesMax; i++ )
+    if ( type == CV_32FC2 )
     {
-        LinePolar line;
-        int idx = _sort_buf[i];
-        int n = cvFloor(idx*scale) - 1;
-        int r = idx - (n+1)*(numrho+2) - 1;
-        line.rho = (r - (numrho - 1)*0.5f) * rho;
-        line.angle = static_cast<float>(min_theta) + n * theta;
-        lines.push_back(Vec2f(line.rho, line.angle));
+        lines = Mat::zeros(linesMax, 1, CV_32FC2);
```

Comment:
This doesn't update/propagate to original "OutputArray". It replaces Mat only.
You should pass OutputArray into this function and call ".create()" to change size.

Reply:
Thank you for the follow. OK, I will fix it like "HoughCircles".

Reply:
I modified to pass OutputArray on function interface.Please check it when you have time.